src/server/server.js

```diff
@@ -124,13 +122,29 @@ app.post('/callback', (req, res) => {
         });
 });
 
+// Protected routes
+app.use('/*', (req, res, next) => {
+    if (
+        process.env.NODE_ENV === 'development' ||
+        authsupport.stillValid(req.session.spadeToken)
+    ) {
+        next();
+    } else {
+        console.log(
+            `no valid session found for ${req.headers['x-forwarded-for'] ||
+                req.connection.remoteAddress}`
+        );
+        res.redirect('/login');
```

Comment:
Vil dette funke på et ajax-kall ?    Og _hvis_ det funker, så burde en kanskje gi klienten muligheten til å lagre unna eventuell state før redirigeringen ?   Kanskje ikke så viktig i speil ennå, men når det blir en fullverdig saksbehandlingsløsning hvor f.eks. saksbehandler i værste fall har komponert et brev som skal lagres :)

Reply:
Jeg vil ikke tro at en redirect for et XHR-kall vil få nettleseren til å sende brukeren til /login, men det gir heller ingen mening å gjøre dette for XHR-kall. Vi får tenke ut noe annet lurt for saksbehandlerbiten.